pkg/contracts/solana/instruction.go

```diff
@@ -582,3 +583,30 @@ func DecodeExecuteMsg(msgbz []byte) (ExecuteMsg, error) {
 
 	return msg, nil
 }
+
+// ProgramInvokedAfterTargetInErrStr checks if any program (other than the specified one)
+// was invoked after the given program in the error string logs.
+func ProgramInvokedAfterTargetInErrStr(errStr, targetProgram string) bool {
```

Comment:
Rather than pure string matching maybe we could type cast it to `*jsonrpc.RPCError` and look at the `Data["logs"]` attribute?

Reply:
want to confirm: the gateway will be passed to `targetProgram` and we're not checking receiver program in here?

Reply:
https://github.com/gagliardetto/solana-go/blob/main/rpc/jsonrpc/jsonrpc.go#L214 is the definitive type of the error returned. You can see it has the structure I mentioned.

Reply:
There will probably have to still be some string parsing but it would be nice to be as targeted as possible.